lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -174,23 +174,19 @@ export default class BitcoinProcessor {
       }
     }
 
-    // wait until at least one block is processed
-    if (this.lastProcessedBlock === undefined) {
-      return {
-        transactions: [],
-        moreTransactions: false
-      };
-    }
-
     console.info(`Returning transactions since ${since ? 'block ' + TransactionNumber.getBlockNumber(since) : 'beginning'}...`);
+    const currentLastProcessedBlock = this.lastProcessedBlock!;
+    let [transactions, numOfBlocksAcquired] = await this.getTransactionsSince(since, currentLastProcessedBlock);
```

Comment:
Design: I am not digging the `lastProcessedBlock` is passed in for no obvious reason externally. Is it possible to let the method assign a local variable and return it out as part of a "tripple" value instead of "tuple"?

Reply:
Using Ali's model. Pass in as the stopping point